src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/InfiniteScrollList.xaml.cs

```diff
@@ -69,6 +69,7 @@ internal InfiniteScrollList(Lazy<JoinableTaskFactory> joinableTaskFactory)
             _joinableTaskFactory = joinableTaskFactory;
 
             InitializeComponent();
+            Unloaded += InfinitScrollListUnloaded;
```

Comment:
I am afraid I am not sure what we need to listen for an unloaded event. 
Wouldn't this get GCed eventually anyways?

If it doesn't get cleaned up, to me this would suggest that there are other potential issues around here.

Reply:
I just followed "Retention path" from memory dump. I am sure there are other problems with XAML code code. But I have very basic experience with XMAL, can't comment what would be other issues.

Reply:
My point was that this would eventually be garbage collected right? 

If you keep open/close the PM UI, you will not run into an OOM will you? 

To me that'd suggest that GC hasn't happened yet because it didn't need to.

Reply:
GC was not able to collect `NuGetVersion`, `VersionRange` instances well. I did both forced GC using tool or waited quite while for eventual GC `with` and `without` my change. Without my changes even after GC, there is always some weak references stays. 

![image](https://user-images.githubusercontent.com/8766776/86942295-d1f36380-c0f9-11ea-9c1f-593f531b4e26.png)

I don't believe I run into  OOM, but if I wait long enough I see memory decrease and GC flag blink on diagnostic tool.
But above is not reliable, so I used forced GC which is quite reliable. If you have any doubt I can demo using screenshare.

Reply:
The event is unsubscribed on line 726, if that's what you're concerned about @nkolev92.

For larger context, subscribing to events and never unsubscribing can lead to memory leaks because of the way that events are implemented and the way that garbage collection works. Depending on how much leaks you might need to open and close the window millions of times to cause an OOM crash, but not unsubscribing can lead to memory leaks.

As long as the Unloaded even is guaranteed to run always, no mater what scenario, there's no potential for leaks.  A more safe/understandable way is to unsubscribe in this classes's dispose method instead, because then we know that it will always be called, no matter what.

Reply:
If something didn't get GCed that means we had a leak. That's what I was trying to understand. 

Do we know exactly why Version/VersionRange weren't getting GCed?
Something in the list or model?

Sounds like we had a problem, but I want to understand exactly where the problem is/was.

Reply:
@nkolev92 As I said before problem is something to do with PM UI XMAL code. 
Maybe we better to pass to @scoban or @mirkomuggli since they're expert on this.